components/ui/project-form.tsx

```diff
@@ -11,56 +11,80 @@ import { useZodForm } from '@/hooks/use-zod-form';
 
 import { addProject, editProject } from '@/app/projects/actions';
 
+import { type Project } from '@prisma/client';
+
 interface Props {
   defaultValues?: Project;
 }
 
-export const schema = z.object({
+const schema = z.object({
   id: z.string().optional(),
   name: z.string().min(1, { message: 'Please enter project name' }),
   upworkId: z.string().nullable(),
   hubstaffId: z.string().nullable(),
   asanaId: z.string().nullable(),
 });
 
-export default function ProjectForm({ defaultValues }: Props) {
+export default function AddProjectForm({ defaultValues }: Props) {
+  const [isPending, startTransition] = useTransition();
+
+  const { execute: executeAdd, status: addStatus } = useAction(addProject, {
+    onSuccess: (data) => {
+      if (data && 'failure' in data) {
+        console.log(data.failure);
+        return;
+      }
```

Comment:
This is artificial code, you only have to write this because of how you designed your server code